### ISSUE TITLE
Panicked at ''range start index 10 out of range for slice of length 7'

### ISSUE THREAD

**Issue by yantene** [lotabout/skim#356]
- distro: archlinux
- version: v0.9.0

After updating the skim version, I'm getting an error when doing the following:

```bash
export SKIM_DEFAULT_OPTIONS='
  --ansi
  --reverse
  --color=matched:0,matched_bg:3,current_bg:8,current_match:0,current_match_bg:3
'

fd -Hc always . | sk
```

The details of the error are as follows.

```
thread 'main' panicked at 'range start index 10 out of range for slice of length 7', src/ansi.rs:430:32
```

It depends on the directory structure of the current directory.

Is there anything I can do?

**Comment by timsampsoncujo** [comment]
I came here to file the same bug. Steps to repro are basically something very similar. Run the following:

`[command that outputs text including ansi colour codes] | sk --ansi` and then hit 'w'

=>
```
  0/41                                                                                                                                                                                                                                                                                                                                                                                      0/0.
> wthread 'main' panicked at 'range start index 2 out of range for slice of length 1', /home/tsampson/.cargo/registry/src/github.com-1ecc6299db9ec823/skim-0.9.0/src/ansi.rs:430:32
stack backtrace:
   0:     0x559c1ee8e82a - std::backtrace_rs::backtrace::libunwind::trace::h622bab51c72c4e69
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/../../backtrace/src/backtrace/libunwind.rs:100:5
   1:     0x559c1ee8e82a - std::backtrace_rs::backtrace::trace_unsynchronized::h7e820b882ebd41ee
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/../../backtrace/src/backtrace/mod.rs:66:5
   2:     0x559c1ee8e82a - std::sys_common::backtrace::_print_fmt::h64d46258114db92f
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/sys_common/backtrace.rs:67:5
   3:     0x559c1ee8e82a - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h47111d0c1b5f0da5
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/sys_common/backtrace.rs:46:22
   4:     0x559c1ed26cdc - core::fmt::write::h1857a60b204f1b6a
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/core/src/fmt/mod.rs:1076:17
   5:     0x559c1ee8df81 - std::io::Write::write_fmt::h4604516fed3e5731
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/io/mod.rs:1516:15
   6:     0x559c1ee8d87e - std::sys_common::backtrace::_print::h785e7a78d5ef272c
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/sys_common/backtrace.rs:49:5
   7:     0x559c1ee8d87e - std::sys_common::backtrace::print::h108047ac5c4555d5
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/sys_common/backtrace.rs:36:9
   8:     0x559c1ee8d87e - std::panicking::default_hook::{{closure}}::h1d0c271f0d087ebf
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/panicking.rs:208:50
   9:     0x559c1ee8d87e - std::panicking::default_hook::h692753e26f11b7b7
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/panicking.rs:227:9
  10:     0x559c1ee8d87e - std::panicking::rust_panic_with_hook::h74ddc20305301cd9
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/panicking.rs:577:17
  11:     0x559c1eea9b28 - std::panicking::begin_panic_handler::{{closure}}::h1a9ca6427ceee1d0
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/panicking.rs:484:9
  12:     0x559c1eea9af4 - std::sys_common::backtrace::__rust_end_short_backtrace::h97e27b7ab754adde
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/sys_common/backtrace.rs:141:18
  13:     0x559c1eea9aad - rust_begin_unwind
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/std/src/panicking.rs:483:5
  14:     0x559c1ed244d0 - core::panicking::panic_fmt::hb15d6f55e8472f62
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/core/src/panicking.rs:85:14
  15:     0x559c1ed26df1 - core::slice::index::slice_start_index_len_fail::hac4fa9c03f14138b
                               at /rustc/e3051d8c24467b54c81da9d9ad70a3e82e106ec1/library/core/src/slice/index.rs:34:5
  16:     0x559c1ee84fcc - <skim::helper::item::DefaultSkimItem as skim::SkimItem>::display::ha5a11a0312272ea2
  17:     0x559c1ee70bdd - <&T as tuikit::draw::Draw>::draw::hfcad844198791bbd
  18:     0x559c1ee4d625 - <tuikit::widget::win::Win<Message> as tuikit::draw::Draw>::draw::h153654c859178a1b
  19:     0x559c1ee77c25 - <tuikit::widget::split::VSplit<Message> as tuikit::draw::Draw>::draw::h34e680a963ad0c32
  20:     0x559c1ee4d625 - <tuikit::widget::win::Win<Message> as tuikit::draw::Draw>::draw::h153654c859178a1b
  21:     0x559c1ee66189 - skim::model::Model::start::h819081fdffe25106
  22:     0x559c1ee1d217 - skim::Skim::run_with::hd1a729acdf753712
  23:     0x559c1edf5c52 - sk::real_main::h423a5e70c126b410
  24:     0x559c1edddd29 - sk::main::hd95effce1032559c
  25:     0x559c1edd5863 - std::sys_common::backtrace::__rust_begin_short_backtrace::h65c83305179266b3
  26:     0x559c1edddab5 - main
  27:     0x7f29db6950b3 - __libc_start_main
  28:     0x559c1eccf09e - _start
  29:                0x0 - <unknown>
```

**Comment by lotabout** [comment]
Sorry for causing trouble. It's been fixed on master by commit: 7f3c73c. I'll file a new release soon.
